lib/extensions/ar_miq_set.rb

```diff
@@ -37,38 +36,53 @@ def make_memberof(set)
   end
 end # module ActsAsMiqSetMember
 
+module ActiveRecord
+  class Base
+    def self.acts_as_miq_set_member
+      include ActsAsMiqSetMember
+    end
+
+    def self.acts_as_miq_set(model_class = nil)
+      include ActsAsMiqSet
+
+      self.model_class = model_class unless model_class.nil?
+    end
+  end
+end
+
 module ActsAsMiqSet
   extend ActiveSupport::Concern
   included do
-    include RelationshipMixin
-    self.default_relationship_type ||= "membership"
-
     include UuidMixin
 
     self.table_name         = "miq_sets"
     self.inheritance_column = :set_type
 
     serialize :set_data
 
-    validates_presence_of     :name
-    validates_uniqueness_of   :name,
-                              :scope => [:set_type, :userid, :group_id],
-                              :if    => proc { |c| c.class.in_my_region.exists?(:name => c.name) }
-    validates_presence_of     :description
-
-    belongs_to  :owner, :polymorphic => true
+    validates :name,
+              :presence                => true,
+              :uniqueness_when_changed => {
+                :scope => [:set_type, :userid, :group_id],
+                :if    => proc { |c| c.class.in_my_region.exists?(:name => c.name) }
+              }
+    validates :description,
+              :presence => true
+
+    belongs_to :owner,
+               :polymorphic => true
+    has_many   :miq_set_memberships,
+               :foreign_key => :miq_set_id,
+               :dependent   => :delete_all
 
     acts_as_miq_taggable
 
-    alias_with_relationship_type :members,            :children
-    alias_with_relationship_type :miq_sets,           :children
-    alias_with_relationship_type :remove_member,      :remove_child
-    alias_with_relationship_type :remove_all_members, :remove_all_children
-    alias_with_relationship_type :add_members,        :add_children
```

Comment:
I'm pretty sure the only callers of `alias_with_relationship_type` are in this file, so when we drop these we can also drop that method

https://github.com/search?q=org%3AManageIQ+alias_with_relationship_type&type=code

Reply:
Noice!  I am on board with that!